src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts

```diff
@@ -109,7 +109,7 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 				});
 				editAction.withAdditionalMeasurements({
 					'elapsedTimeMs': new Date().getTime() - startTime
-				}).send();
+				}).withAdditionalProperties(result.propertyBag).send();
```

Comment:
Well yes, but custom telemetry properties are inherently provider specific anyways

Reply:
I was thinking if we make it clear the metadata is going to be included in the telemetries so that we don't have to have provider specific information in ADS core, but I kind of agree with you, it is kind of risky to trust the extension 100%. 

if we go with your approach, we have to keep an allow list per provider (a single allow list for all providers won't work). If extension want to include their metadata in telemetry, they will have to create a PR to update azuredatastudio code to include their properties. is this what you were envisioning?

Reply:
If you're doing it per-provider though why not have the telemetry live on the provider side? Then they are free to add whatever data they want and we may not even need to be adding custom properties to the events like this.

Reply:
@abist does it make sense to you? let us know if you have questions.

Reply:
it will still be being passed in by extension, let me describe what I am thinking more clearly.
1. rename the propertyBag to metadata in the typings file and add description.
2. add a new file in table designer folder in core with a mapping of provider and the allow list of property names to be sent with telemetry for it.
3. when sending telemetry, inspect metadata property and only include the properties listed in the allow list for current provider.

Reply:
Makes sense. So adding a method such as `sendMetadata` to the`TableDesignerProvider` to explicitly get the data from the server via a request and then send that data using core telemetry, right?